classes/components/forms/FieldSlider.php

```diff
@@ -0,0 +1,78 @@
+<?php
+/**
+ * @file classes/components/form/FieldSlider.php
+ *
+ * Copyright (c) 2014-2024 Simon Fraser University
+ * Copyright (c) 2000-2024 John Willinsky
+ * Distributed under the GNU GPL v3. For full terms see the file docs/COPYING.
+ *
+ * @class FieldSlider
+ *
+ * @ingroup classes_controllers_form
+ *
+ * @brief A color picker field in a form.
+ */
+
+namespace PKP\components\forms;
+
+class FieldSlider extends Field
+{
+    /** @copydoc Field::$component */
+    public $component = 'field-slider';
+
+
+    /**
+     * Range min value
+     */
+    public int $min;
+
+    /**
+     * Range max value
+     */
+    public int $max;
+
+    /**
+     * Label for min value, it displays actual value when not present
+     */
+    public ?string $minLabel = null;
+
+    /**
+     * Label for max value, it displays actual value when not present
+     */
+    public ?string $maxLabel = null;
+
+    /**
+     * Expects translation string, which should contain {$value} placeholder. It displays actual value if not present.
+     */
+    public ?string $valueLabel = null;
+
+    /**
+     * Expects translation string, which might contain {$value} placeholder. It fallback to valueLabel is not present..
+     */
+    public ?string $valueLabelMin = null;
+
+    /**
+     * Expects translation string, which might contain {$value} placeholder. It fallback to valueLabel is not present..
+     */
+    public ?string $valueLabelMax = null;
+
+
+
+    /**
+     * @copydoc Field::getConfig()
+     */
+    public function getConfig()
+    {
+        $config = parent::getConfig();
+        $config['min'] = $this->min;
+        $config['max'] = $this->max;
+        $config['minLabel'] = $this->minLabel;
+        $config['maxLabel'] = $this->maxLabel;
+        $config['valueLabel'] = $this->valueLabel;
+        $config['valueLabelMin'] = $this->valueLabelMin;
+        $config['valueLabelMax'] = $this->valueLabelMax;
```

Comment:
Along with all these control option , I think we should also add the ability to define what is the range steps would be . It should not be a nullable and allow both int and float value .

Reply:
@touhidurabir On the other hand - the improvements you are suggesting are small effort..  so not really problem include them. Will do the int/float & step property.

Reply:
As the value can be float and in such case steps can be float also . Right now we do no have a use case but it's not unnatural to have the slider range increase like [0.5, 1.0, 1.5, 2.0, 2.5, ...] .